src/taskgraph/run-task/run-task

```diff
@@ -1061,6 +1069,9 @@ def main(args):
     parser.add_argument("--user", default="worker", help="user to run as")
     parser.add_argument("--group", default="worker", help="group to run as")
     parser.add_argument("--task-cwd", help="directory to run the provided command in")
+    parser.add_argument("--shallow", action="store_true",
+        default=False, help="perform a shallow clone (default: false)",
+    )
```

Comment:
I'm really sorry, I told you to add an argument here. But on closer look, I was wrong and instead we should add this in `collect_vcs_options` using an environment variable like most of the other vcs arguments. You should add something like:
```
shallow = os.environ.get("%s_SHALLOW_CLONE" % env_prefix)
```

and make sure to return it (like you already are).